src/leo_s3_auth.erl

```diff
@@ -359,6 +455,65 @@ setup(DB, Provider) ->
     true = ets:insert(?AUTH_INFO, {1, #auth_info{db       = DB,
                                                  provider = Provider}}),
     ok.
+%% @doc Trim Space From Binary
+%% @private
+trim(<<>>) ->
+    <<>>;
+trim(Bin) ->
+    trim(Bin, byte_size(Bin)).
+
+trim(Bin = <<Byte:1/binary, Tail/binary>>, Size) ->
+    case is_space(Byte) of
+        true ->
+            trim(Tail, Size - 1);
+        false ->
+            trim_tail(Bin, Size)
+    end.
+
+trim_tail(Bin, Size) ->
+    SizeMinus1 = Size - 1,
+    <<Rest:SizeMinus1/binary, Byte:1/binary>> = Bin,
+    case is_space(Byte) of
+        true ->
+            trim_tail(Rest, Size - 1);
+        false ->
+            Bin
+    end.
+
+is_space(<<" ">>) ->
+    true;
+is_space(_) ->
+    false.
```

Comment:
and also since this function triming binary is reusable, 
please add to leo_commons.